src/slip.rs

```diff
@@ -0,0 +1,200 @@
+use secp256k1::{PublicKey};
+
+/// A UTXO slip containing record of owernship of funds on the network
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Slip {
+    /// Contains concrete data which is not relative to state of the chain
+    body: SlipBody,
+    /// Flag determining if the `Slip` is part of the longest chain
+    lc: bool,
+    /// `SlipSpentStatus` indicating the state of the slip in the network
+    pub spent_status: SlipSpentStatus,
+}
+
+/// An object that holds concrete data not subjective to state of chain
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct SlipBody {
+    /// An `Sectp256K::PublicKey` determining who owns the `Slip`
+    add: PublicKey,
+    /// A enum brodcast type determining how to be processed by consensus
+    typ: SlipBroadcastType,
+    /// Amount of Saito
+    amt: u64,
+    /// `Block` id
+    bid: u64,
+    /// `Transaction` id
+    tid: u64,
+    /// `Slip` id
+    sid: u64,
+    /// The `Block` hash the slip originated from
+    bsh: [u8; 32],
+}
+
+/// An enumerated set of `Slip` types
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum SlipBroadcastType {
+    Normal,
+    GoldenTicket,
+    Fee,
+    Rebroadcast,
+    VIP,
+    GoldenChunk,
+}
+
+/// Current status of the `Slip` state with regards to it's usage in the network
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum SlipSpentStatus {
+    Unspent,
+    Spent,
+    Pending,
+}
+
+impl Slip {
+    /// Create new `Slip` with default type `SlipBroadcastType::Normal`
+    ///
+    /// * `add` - `Publickey` address to assign ownership
+    /// * `typ` - `SlipBroadcastType` of `Slip`
+    /// * `amt` - `u64` amount of Saito coantined in the `Slip`
+    pub fn new(add: PublicKey, typ: SlipBroadcastType, amt: u64) -> Slip {
+        return Slip {
+            body: SlipBody {
+                add,
```

Comment:
This is an issue we've already resolved, although I agree with you escapedcat. There is a legitimate reason in saito-lite to keep these small because we are using json-encoded blocks and there is an argument that we should keep things looking similar to saito-lite. I personally don't agree with this reasoning but find it acceptable so long as we add comments giving them more expressive names like has been done above.

Reply:
I think it will be more troublesome to rename those ambiguous down the road so its better to rename when its still early. Adding comments is not going to make it more understandable when its used in modules.

Reply:
I agree with that. If it doesn't become an issue increasing the size when encoding slips, having more specific names is clearly superior

Reply:
In `bincode`, attribute naming does not affect the encoded size so I think we can go with more specific names.

Reply:
I agree with batzor on this.

Reply:
Stephen is working on another branch where we can test bincode vs protobufs vs whatever else and then have some more objective conversation about this.